app/components/NoticeTypeCheckboxes.tsx

```diff
@@ -157,29 +159,161 @@ const NoticeTypes = {
   ],
 }
 
-interface NoticeTypeCheckboxeProps {
+type DataRow = {
+  topic: string
+  latestRate: number
+}
+
+type FormattedData = {
+  text: DataRow[]
+  binary: DataRow[]
+  voevent: DataRow[]
+}
+
+interface NoticeTypeCheckboxProps {
   defaultSelected?: string[]
+  selectedFormat?: 'binary' | 'text' | 'voevent'
+  showTotalSelected?: boolean
+}
+
+export async function loadTriggerRateData() {
+  return await getAverageTriggerRate()
 }
 
 export function NoticeTypeCheckboxes({
   defaultSelected,
-}: NoticeTypeCheckboxeProps) {
+  selectedFormat = 'text',
+  showTotalSelected = false,
+}: NoticeTypeCheckboxProps) {
+  const [metrics, setMetrics] = useState<FormattedData>()
+  const [userSelected, setUserSelected] = useState<{ [key: string]: boolean }>(
+    {}
+  )
+  const [selectedCounter, setSelectedCounter] = useState(0)
+  const [alertEstimate, setAlertEstimate] = useState(0)
+  useEffect(() => {
+    loadTriggerRateData().then((response) => {
+      setMetrics({
+        text: response.result
+          .filter(
+            (item: { metric: { topic: string } }) =>
+              item.metric.topic.indexOf('.text.') != -1
+          )
+          .map((item: { metric: { topic: any }; values: string | any[] }) => ({
+            topic: item.metric.topic.split('.')[3],
+            latestRate: Math.ceil(item.values[item.values.length - 1][1]),
+          })),
+        binary: response.result
+          .filter(
+            (item: { metric: { topic: string } }) =>
+              item.metric.topic.indexOf('.binary.') != -1
+          )
+          .map((item: { metric: { topic: any }; values: string | any[] }) => ({
+            topic: item.metric.topic.split('.')[3],
+            latestRate: Math.ceil(item.values[item.values.length - 1][1]),
+          })),
+        voevent: response.result
+          .filter(
+            (item: { metric: { topic: string } }) =>
+              item.metric.topic.indexOf('.voevent.') != -1
+          )
+          .map((item: { metric: { topic: any }; values: string | any[] }) => ({
+            topic: item.metric.topic.split('.')[3],
+            latestRate: Math.ceil(item.values[item.values.length - 1][1]),
+          })),
+      })
+    })
+  }, [])
+
+  const counterfunction = (childRef: HTMLInputElement) => {
+    userSelected[childRef.name] = childRef.checked
+    setUserSelected(userSelected)
+
+    let selectedTotal = 0
+    for (const key of Object.keys(userSelected)) {
+      if (userSelected[key]) {
+        selectedTotal++
+      }
+    }
+    setSelectedCounter(selectedTotal)
+
+    if (metrics) {
+      setAlertEstimate(getSum(metrics))
+    }
+  }
+
+  function getSum(metrics: FormattedData) {
+    return metrics[selectedFormat]
+      .filter(
+        (item) =>
+          userSelected[item.topic] !== undefined &&
+          userSelected[item.topic] == true
```

Comment:
Shouldn't that be equivalent to this?

```suggestion
          userSelected[item.topic]
```

or this?

```suggestion
          !!userSelected[item.topic]
```

Reply:
yup! updated in latest!